plugins/plugins.go

```diff
@@ -395,8 +405,10 @@ func (m *Manager) Stop(ctx context.Context) {
 		}
 	}()
 
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Duration(m.gracefulShutdownPeriod)*time.Second)
+	defer cancel()
```

Comment:
So this `defer cancel()` will be a problem for any goroutine spawned off the `Stop(ctx)` calls: _those_ contexts would be cancelled when _this method_ returns, and that may very well be too early, before the timeout.

Reply:
My assumption here was that each `Stop` call is blocking, and done with whatever it needs to stop the plugin when returning - if that assumption doesn't hold, how do we know when a plugin has _really_ stopped?

Stopping well before the timeout is of course fine if all plugins have reported back as done with their Stop processes.

Reply:
💡 Ah, so I've misread the logs flushing code. Of course it blocks on `ctx.Done()` 🤦 

Even better, it means that we can call `cancel` here, or not call it, it won't matter. If the timeout has occurred, it will make no difference, if it hasn't, noone is using that ctx anymore.

Reply:
Indeed - using/calling cancel here is mainly to please go-check or whatever tool it is that keeps complaining if not doing that 😄